Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ExchangeManagerBean.java

```diff
@@ -329,13 +329,23 @@ public boolean deleteExchange() {
             if (bDelete) {
                 modifiedExchangesCache();
                 selectedExchange = null;
+                filterExchange = null;
+                filterOrganization = null;
+                orgFilter = null;
+                endpoints = null;
+                refreshOrganizations();
             }
         }
         return bDelete;
     }
 
     public int pingEndpoint() {
-        return exchangeService.pingService(selectedEndpoint);
+        for (ConnectionEndpoint connEndpoint : endpoints) {
```

Comment:
Can the endpoints be null here?

Reply:
No, because it's using an existing list to select the endpoint.  That's why there wasn't a null check for the old code.